### PR TITLE
Add organization column to [Tank, Family, Animal]

### LINKED ISSUE #333: Scope Animal to an Organization
As we plan the multi-tenancy of the application, a Animal should belong to an Organization.

Acceptance Criteria

- [x] Add a belongs_to relationship on Animal
- [x] Add a has_many relationship on Organization
- [x] Add an organization_id column to Animal with an index
- [x] Update existing factory for Animal with an organization
- [x] Add specs to Organization and to Animal for the added relationship
- [x] Scope queries that occur on the `AnimalsController` to the current logged in users organization

### PR DESCRIPTION
**Pull request by haydenrou**
Add organization association to animal
Retofit specs to accommodate organizations for the csv importer.
Update the animals controller to scope using the new
Add a concern for OrganizationScope which can be added to any model with
an organization association

Resolves #333 #334 #336

### Type of change

* Bug fix (non-breaking change which fixes an issue)
* New feature (non-breaking change which adds functionality)
* Breaking change (fix or feature that would cause existing functionality to not 
work as expected)
* This change requires a documentation update
* Documentation update

### CODE REVIEW

**Comment by haydenrou** [comment]
Hi @benreyn we are looking to address #336 #334 #333 in this PR. We have only started #333 here and wouldn't mind some feedback on if you are happy with the approach we are taking before pushing the rest up.

As we are adding foreign keys here, we'd be interested to know if these tables are being used yet, so as whether we need to write a back-fill migration to fill the `organization_id`'s in for each of these models.

**Comment by benreyn** [comment]
> we'd be interested to know if these tables are being used yet, so as whether we need to write a back-fill migration to fill the organization_id's in for each of these models. 

No need for this, as we haven't reached our MVP yet, we are safely able to nuke the database and start fresh. One thing to note though is that the db:seed tasks might need to be updated.

> Additionally, would you be opposed to us adding organization_id to many other models? We were thinking of streamlining this across the application's models more than just the three aforementioned ones.

Curious which models you have in mind. (note that many of the db tables are actually dead code that we need to eventually remove, See [these](https://github.com/rubyforgood/abalone/issues?q=is%3Aissue+is%3Aopen+label%3A%22Technical+Debt%22) issues if your curious about which ones.

The only models that we will need to account for this for in the MVP are `Tank`, `Facility`, `Animal`, `Family`, `Measurement`, `Operation`, `Measurement` and `MeasurementEvent`.

@CraigJZ Did I miss anything there?

@haydenrou, if there are models that we have missed in that list above, then yes, we do want to make sure that they are also organization level scoped.

**Comment by CraigJZ** [comment]
@benreyn No, I think that covers it. Thanks!!

File: app/controllers/animals_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class AnimalsController < ApplicationController
   before_action :set_animal, only: [:show, :edit, :update, :destroy]
 
   def index
-    @animals = Animal.all
+    @animals = Animal.for_organization(current_user.organization)
```
Comment (haydenrou) [inline app/controllers/animals_controller.rb:5]: `current_organization`

**Review by CraigJZ — COMMENTED** [review]
Sorry for the delay in getting to this. I think we're good here but just need to resolve the merge conflict.

Thanks!!!!

**Review by todtb — APPROVED** [review]

**Pull request merged**